docs/HANDLERS.md

```diff
@@ -0,0 +1,376 @@
+# Handlers
+
+In addition to the Applications the toolkit provides a variety of the utilities to perform most common actions of the ship. Following list of handlers includes a prebuild Express routers which performs most common operations of connectors. 
+
+### actionHandler() `incoming`
+This is the simplest requests handler to expose custom logic through custom API POST endpoint. The possible usage is triggering a custom operation (like fetching historical data) or a webhook. Both cases handle data `incoming` into Hull platform. However in case of busy webhook it's better to use `batcherHandler` which automatically group the incoming requests into batches.
+
+```js
+const app = new Hull.App({}).server();
+import { actionHandler } from "hull/lib/utils";
+
+app.use("/fetch-all", actionHandler((ctx, { query, body }) => {
+  const { hull, ship } = ctx;
+
+  const { api_token } = ship.private_settings;
+  const serviceClient = new ServiceClient(api_token);
+  return serviceClient.getHistoricalData()
+    .then(users => {
+      users.map(u => {
+        hull.as({ email: u.email }).traits({
+          new_trait: u.custom_value
+        });
+      });
+    });
+}));
+```
+
+### batcherHandler() `incoming`
+The second `incoming` handler which works in a simillar way as `actionHandler` but it also groups incoming requests into batches of selected size:
+
+```js
+const app = new Hull.App({}).server();
+import { batcherHandler } from "hull/lib/utils";
+
+app.use("/fetch-all", batcherHandler((ctx, requests) => {
+  requests.map(request => {
+    console.log(request); // { query, body }
+  })
+}, {
+  maxSize: 100, // maximum size of the batch
+  maxTime: 1000 // time time in milliseconds to flush batch after the first item came in
+}));
+```
+
+### notifHandler() `outgoing`
```

Comment:
Yes, I was thinking about that, I would really like to use the terms `incoming` and `outgoing` in context of Hull platform and ecosystem in general, exactly the same how we understand them in the manifest.json tags.

I know that from connector perspective it's more complex, so we have incoming data of outgoing flow, like here in `notifHandler`, but to avoid any confusion I want to have a "foundation/base" terms so every ship developer know where he is at the higher level.

Maybe we can come up with additional terms to call this lower level flow?